libraries/rush-lib/src/utilities/prompts/SearchListPrompt.ts

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { Interface } from 'readline';
+import _ from 'lodash';
+import colors from 'colors/safe';
+
+// Modified from the choice list prompt in inquirer:
+// https://github.com/SBoudrias/Inquirer.js/blob/inquirer%407.3.3/packages/inquirer/lib/prompts/list.js
+// Extended to include text filtering for the list
+import type inquirer from 'inquirer';
+import type { Answers, ListQuestion, DistinctChoice } from 'inquirer';
+import BasePrompt from 'inquirer/lib/prompts/base';
+import observe from 'inquirer/lib/utils/events';
+import Paginator from 'inquirer/lib/utils/paginator';
+import type Separator from 'inquirer/lib/objects/separator';
+import type Choice from 'inquirer/lib/objects/choice';
+import type Choices from 'inquirer/lib/objects/choices';
+
+import figures from 'figures';
+
+const { map, takeUntil } = require('rxjs/operators');
+
+interface IKeyPressEvent {
+  key: { name: string; ctrl: boolean; sequence?: string };
+}
+
+export class SearchListPrompt extends BasePrompt<ListQuestion> {
+  protected done!: (result: unknown) => void;
+
+  private readonly _paginator: Paginator;
+  private _selected: number = 0;
+  private _query: string = '';
+  private _firstRender: boolean = true;
+
+  public constructor(question: ListQuestion, readline: Interface, answers: Answers) {
+    super(question, readline, answers);
+
+    if (!this.opt.choices) {
+      this.throwParamError('choices');
+    }
+
+    if (
```

Comment:
Note that you don't need to implement default if you aren't using it

Reply:
Noted, but I like keeping the implementation as close to default as possible from the original ListPrompt.